core/src/components/datetime/datetime.tsx

```diff
@@ -528,10 +563,49 @@ export class Datetime implements ComponentInterface {
     };
   };
 
-  private setActiveParts = (parts: DatetimeParts) => {
-    this.activeParts = {
-      ...parts,
-    };
+  private setActiveParts = (parts: DatetimeParts, removeDate = false) => {
+    const { multiple, activePartsClone, highlightActiveParts } = this;
+
+    if (multiple) {
+      /**
+       * We read from activePartsClone here because valueChanged() only updates that,
+       * so it's the more reliable source of truth. If we read from activeParts, then
+       * if you click July 1, manually set the value to July 2, and then click July 3,
+       * the new value would be [July 1, July 3], ignoring the value set.
+       *
+       * We can then pass the new value to activeParts (rather than activePartsClone)
+       * since the clone will be updated automatically by activePartsChanged().
+       */
+      const activePartsArray = Array.isArray(activePartsClone) ? activePartsClone : [activePartsClone];
```

Comment:
~ We repeat this similar manipulation in multiple places. Would it make sense to extract this behavior in a util that all the implementations share?

```ts
const activePartsArray = valueToArray(activePartsClone);
```

```ts
const valueToArray = (data: string | string[]) => Array.isArray(data) ? data : [data];
```

Reply:
I could, though it's only done here and one other time in `state.ts`; I'm not sure the extra layer of logic is worth the minimal code length difference 🤔